### PR TITLE
update importCosmicData: flexible reading for new version of cosmic

### PR DESCRIPTION
**Pull request by egarcialara**
Fix: import cosmic data with new COSMIC versions. Tested for v.94

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java
Comment (pvannierop) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java:116]:
```suggestion
                    int count = Integer.parseInt(removePrefix(fields[inxCntField]));
```

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java
Comment (pvannierop) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java:85]: What is the reason for removing these fields?

**Review by pvannierop — CHANGES_REQUESTED** [review]
One question and a change request.

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java
Comment (egarcialara) [reply core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java:95]: It is common that in the same file the fields HGVSP, HGVSG, HGVSC and SNP are present only in some rows. This affects the selection after the first row.

Comment (pvannierop) [reply core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java:95]: But the presence of these fields does not interfere with the function of the script, does it? At least I do not see how that is a problem.

Comment (egarcialara) [reply core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java:95]: It does when the column 7 looks like:
```
GENE=; STRAND=; CDS=; HGVSP=; AA=; CNT=
GENE=; STRAND=; CDS=; HGVSP=; AA=; CNT=
GENE=; STRAND=; CDS=; AA=; CNT=
GENE=; STRAND=; CDS=; HGVSPC; HGVSP=; AA=; CNT=
GENE=; STRAND=; CDS=; HGVSP=; HGVSP=; SNP; AA=; CNT=
```
In this case only rows 1 and 2 would be correct.

**Review by pvannierop — APPROVED** [review]

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java
Comment (dippindots) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java:95]: I noticed that we comment this part out currently, could you tell me why do we need to add this back?

**Review by dippindots — COMMENTED** [review]
Thanks @egarcialara. It looks good to me, I just have one question regarding the `gene` field.

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java
Comment (egarcialara) [reply core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java:95]: Hi @dippindots, thank you for reviewing the PR!

I have uncommented it because the new cosmic files have genes with that prefix (e.g. `GENE=PLCH2_ENST00000378488`), and this code allows to insert the record instead of getting the message `Gene symbol in COSMIC not recognized: PLCH2_ENST00000378488`.

**Review by dippindots — APPROVED** [review]

**Comment by dippindots** [comment]
@egarcialara @pvannierop Last question before we merge it, is this pr backward compatible? If we merge it, can we still import the old version of cosmic data?

**Comment by egarcialara** [comment]
> @egarcialara @pvannierop Last question before we merge it, is this pr backward compatible? If we merge it, can we still import the old version of cosmic data?

Hi @dippindots, yes this can import the old version. We just made the reading more flexible.

**Comment by dippindots** [comment]
@egarcialara Thanks, got it. Could you please squash all the commit into one and rebase to the latest master? I will merge after that.

**Comment by egarcialara** [comment]
@dippindots Done!

**Pull request merged**
